dipy/viz/app.py

```diff
@@ -465,188 +451,264 @@ def win_callback(obj, event):
-                            t = self.cla[bundle]['tractogram']
-                            c = self.cla[bundle]['cluster']
-                            indices = self.tractogram_clusters[t][c]
-                            active_streamlines.extend(Streamlines(indices))
-
-                    # self.tractograms = [active_streamlines]
-                    hz2 = Horizon([active_streamlines],
-                                  self.images, cluster=True, cluster_thr=5,
-                                  random_colors=self.random_colors,
-                                  length_lt=np.inf,
-                                  length_gt=0, clusters_lt=np.inf,
-                                  clusters_gt=0,
-                                  world_coords=True,
-                                  interactive=True)
-                    ren2 = hz2.build_scene()
-                    hz2.build_show(ren2)
+                    new_window()
 
                 if key == 'a' or key == 'A':
-
-                    if self.select_all is False:
-                        for ca in self.cea:
-                            if (self.cea[ca]['length'] >= self.length_min and
-                                    self.cea[ca]['size'] >= self.size_min):
-                                self.cea[ca]['selected'] = 1
-                                cas = self.cea[ca]['cluster_actor']
-                                self.cla[cas]['selected'] = \
-                                    self.cea[ca]['selected']
-                        self.show_m.render()
-                        self.select_all = True
-                    else:
-                        for ca in self.cea:
-                            if (self.cea[ca]['length'] >= self.length_min and
-                                    self.cea[ca]['size'] >= self.size_min):
-                                self.cea[ca]['selected'] = 0
-                                cas = self.cea[ca]['cluster_actor']
-                                self.cla[cas]['selected'] = \
-                                    self.cea[ca]['selected']
-                        self.show_m.render()
-                        self.select_all = False
+                    show_all()
 
                 if key == 'e' or key == 'E':
+                    expand()
 
-                    for c in self.cea:
-                        if self.cea[c]['selected']:
-                            if not self.cea[c]['expanded']:
-                                len_ = self.cea[c]['length']
-                                sz_ = self.cea[c]['size']
-                                if (len_ >= self.length_min and
-                                        sz_ >= self.size_min):
-                                    self.cea[c]['cluster_actor']. \
-                                        VisibilityOn()
-                                    c.VisibilityOff()
-                                    self.cea[c]['expanded'] = 1
+                if key == 'r' or key == 'R':
```

Comment:
`reset()` unhide hidden clusters?  it will be good to collapse only the visible cluster